[BUG] Vercel deployment fails with ERR_MODULE_NOT_FOUND for dynamically generated tool files

### Describe the Bug

## Description

When deploying a Mastra application to Vercel, the deployment fails with the following error:

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/tools/30caf52b-d2b0-42ca-8c37-e59c7b36b3b8.mjs' imported from /var/task/index.mjs
at finalizeResolution (node:internal/modules/esm/resolve:281:11)
...
```

## Cause

The issue is in the generated `vercel.json` configuration. When specifying the `"tools"` directory in the `includeFiles` array, Vercel does not include all files within that directory, especially dynamically generated files with UUID patterns.

## Solution

Change the `includeFiles` configuration in `vercel.json` from:

```json
{
  "version": 2,
  "installCommand": "npm install --omit=dev",
  "builds": [
    {
      "src": "index.mjs",
      "use": "@vercel/node",
      "config": {
        "includeFiles": [
          "index.mjs",
          "index2.mjs",
          "instrumentation.mjs",
          "mastra.mjs",
          "node_modules",
          "package-lock.json",
          "package.json",
          "telemetry-config.mjs",
          "tools",
          "tools.mjs"
        ]
      }
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "index.mjs"
    }
  ]
}
```

To:
```json
{
  "version": 2,
  "installCommand": "npm install --omit=dev",
  "builds": [
    {
      "src": "index.mjs",
      "use": "@vercel/node",
      "config": {
        "includeFiles": [
          "index.mjs",
          "index2.mjs",
          "instrumentation.mjs",
          "mastra.mjs",
          "node_modules",
          "package-lock.json",
          "package.json",
          "telemetry-config.mjs",
          "tools/**",
          "tools.mjs"
        ]
      }
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "index.mjs"
    }
  ]
}
```

### Steps To Reproduce

1. Create mastra with `npx create-mastra@latest`
2. npm install @mastra/deployer-vercel
3. Edit the src/mastra/index.ts to use VercelDeployer
4. mastra build
5. cd .mastra/output
6. vercel --prod

Setup selection
```
Need to install the following packages:
create-mastra@0.3.0
Ok to proceed? (y) y

┌  Mastra Create
│
◇  What do you want to name your project?
│  vercel-deployer-issue-test
│
◇  Project created
│
◇  npm dependencies installed
│
◇  mastra installed
│
◇  Dependencies installed
│
◇  .gitignore added
│
└  Project created successfully


┌  Mastra Init
│
◇  Where should we create the Mastra files? (default: src/)
│  src/
│
◇  Choose components to install:
│  Agents
│
◇  Add tools?
│  Yes
│
◇  Select default provider:
│  OpenAI
│
◇  Enter your openai API key?
│  Skip for now
│
◇  Add example
│  Yes
│
◇  Make your AI IDE into a Mastra expert? (installs Mastra docs MCP server)
│  Skip for now
```

### Expected Behavior

The API server should start without runtime error.

### Environment Information

- Mastra version: 0.6.0
- node v23.10.0 (npm v10.9.2)
- macOS 14.3.1（23D60）

### Component

Deployment

### Code Snippet

```typescript
src/mastra/index.ts

import { Mastra } from '@mastra/core/mastra';
import { createLogger } from '@mastra/core/logger';
import { LibSQLStore } from '@mastra/libsql';

import { weatherAgent } from './agents';
import { VercelDeployer } from '@mastra/deployer-vercel';

const vercelDeployer = new VercelDeployer({
  teamSlug: 'your-team-slug',
  projectName: 'your-project-name',
  token: 'your-vercel-token',
});

export const mastra = new Mastra({
  agents: { weatherAgent },
  storage: new LibSQLStore({
    // stores telemetry, evals, ... into memory storage, if it needs to persist, change to file:../mastra.db
    url: ":memory:",
  }),
  logger: createLogger({
    name: 'Mastra',
    level: 'info',
  }),
  deployer: vercelDeployer,
});
```

### Error Logs

```shell
1 Error

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/tools/71551509-5e98-477d-aee2-616eef081f08.mjs' imported from /var/task/index.mjs
    at finalizeResolution (node:internal/modules/esm/resolve:281:11)
    at moduleResolve (node:internal/modules/esm/resolve:866:10)
    at moduleResolveWithNodePath (node:internal/modules/esm/resolve:990:14)
    at defaultResolve (node:internal/modules/esm/resolve:1033:79)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:685:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:634:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:617:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:273:38)
    at onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:577:36)
    at TracingChannel.tracePromise (node:diagnostics_channel:344:14) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///var/task/tools/71551509-5e98-477d-aee2-616eef081f08.mjs'
}
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.
```

### Additional Context

_No response_

### Verification

- [x] I have searched the existing issues to make sure this is not a duplicate
- [x] I have included sufficient information for the team to reproduce and understand the issue

--- a/.changeset/kind-days-stand.md
+++ b/.changeset/kind-days-stand.md
@@ -1,0 +1,5 @@
+---
+'@mastra/deployer-vercel': patch
+---
+
+Fix bug where we were not traversing dirs when creating vercel.json to add al files to the includedFiles array

--- a/deployers/vercel/src/index.ts
+++ b/deployers/vercel/src/index.ts
@@ -196,8 +196,15 @@ export const POST = handle(app);
     const result = await this._bundle(this.getEntry(), entryFile, outputDirectory, toolsPaths);
 
     // read dist files one level deep in the output directory
-    const files = readdirSync(join(outputDirectory, this.outputDir));
-    this.writeVercelJSON(outputDirectory, files);
+    const files = readdirSync(join(outputDirectory, this.outputDir), {
+      recursive: true,
+    });
+
+    const filesWithoutNodeModules = files.filter(
+      file => typeof file === 'string' && !file.startsWith('node_modules'),
+    ) as string[];
+
+    this.writeVercelJSON(outputDirectory, filesWithoutNodeModules);
 
     return result;
   }

